src/components/button/button.component.js

```diff
@@ -50,6 +57,12 @@ const Button = (props) => {
     buttonType: buttonType || as
   };
 
+  if (fullWidth && (size !== 'large' || buttonType !== 'dashed')) {
+    const message = 'Buttons fullWidth property can only be used with button'
+    + ' which prop size is `large and buttonType is `dashed`';
+    invariant(size === 'large' && buttonType === 'dashed', message);
```

Comment:
No need to nest this in an `if`
```suggestion
    invariant(!fullWidth || size === 'large' && buttonType === 'dashed', message);
```

Reply:
updated